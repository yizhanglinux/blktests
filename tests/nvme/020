#!/bin/bash
# SPDX-License-Identifier: GPL-2.0+
# Copyright (c) 2017-2018 Western Digital Corporation or its affiliates.
#
# Test NVMe DSM Discard command on NVMeOF with a file-backed ns.

. tests/nvme/rc

DESCRIPTION="test NVMe DSM Discard command on NVMeOF file-backed ns"
QUICK=1

requires() {
	_nvme_requires
	_require_nvme_trtype_is_fabrics
}

test() {
	echo "Running ${TEST_NAME}"

	_setup_nvmet

	local ns
	local nblk_range="10,10,10,10,10,10,10,10,10,10"
	local sblk_range="100,200,300,400,500,600,700,800,900,1000"

	_nvmet_target_setup --blkdev file

	_nvme_connect_subsys

	ns=$(_find_nvme_ns "${def_subsys_uuid}")

	nvme dsm "/dev/${ns}" --ad \
		--slbs "${sblk_range}" --blocks "${nblk_range}"

	_nvme_disconnect_subsys

	_nvmet_target_cleanup

	echo "Test complete"
}
