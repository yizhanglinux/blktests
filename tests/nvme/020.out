Running nvme/020
NVMe DSM: success
disconnected 1 controller(s)
Test complete
